conda_build/skeletons/cran.py

```diff
@@ -1165,6 +1166,18 @@ def skeletonize(in_packages, output_dir=".", output_suffix="", add_maintainer=No
         if 'R' not in dep_dict:
             dep_dict['R'] = ''
 
+        os_type = cran_package.get("OS_type", '')
+        if os_type != 'unix' and os_type != 'windows' and os_type != '':
+            print("Unknown OS_type: {} in CRAN package".format(os_type))
+            os_type = ''
+        if os_type == 'unix':
+            d['skip_os'] = 'skip: True  # [not unix]'
+            d["noarch_generic"] = ""
+        if os_type == 'windows':
+            d['skip_os'] = 'skip: True  # [not win]'
```

Comment:
I think you will need to add the ``d["noarch_generic"] = ""`` here as well
```suggestion
            d['skip_os'] = 'skip: True  # [not win]'
            d["noarch_generic"] = ""
```

If @mingwandroid has a better idea that is fine for me! :)